_posts/2019-08-06-serverless-static-sites.md

```diff
@@ -0,0 +1,232 @@
+```sh
+git clone https://github.com/openfaas-incubator/ingress-operator
+cd ingress-operator
+
+kubectl apply -f ./artifacts/operator-crd.yaml
+kubectl apply -f ./artifacts/operator-rbac.yaml
+kubectl apply -f ./artifacts/operator-amd64.yaml
+```
+
+## Introducing the hugo template
+Today I'm showing you a new template for creating [hugo](https://gohugo.io) static sites that took me little to no time to make thanks to the simplicity and flexibility of [OpenFaaS's templating system](https://www.openfaas.com/blog/template-store/). You can check it out [here](https://github.com/matipan/openfaas-hugo-template).  
+The template copies over the contents of your hugo site, builds it into the `public` directory and then uses a [very lightweight static server](https://gitlab.com/matipan/static-server) that serves the content and provides a healtcheck that follows the standards from OpenFaaS.
+
+## Create a new Hugo site
+We can use the hugo template to create hugo sites and then serve them with custom domain names using OpenFaaS and the Ingress Operator. First lets create the site:
+```sh
+git init
+faas template pull https://github.com/matipan/openfaas-hugo-template
+faas new --lang hugo -g <openfaas gateway url> --prefix <docker hub username> example-site
+```
+This will create a folder called `example-site`, `cd` into it and now create the site with this instructions from the [hugo quickstart guide](https://gohugo.io/getting-started/quick-start/#step-2-create-a-new-site):
+```sh
+hugo new site .
+git submodule add https://github.com/budparr/gohugo-theme-ananke.git themes/ananke
+echo 'theme = "ananke"' >> config.toml
+```
+At this point you can run `hugo server` in that directory to build and test your site locally without needing to deploy it to OpenFaaS. Remember to update the `baseURL` found in the `config.toml` to the domain that you will be using.
+
+Deploy your blog:
+```sh
+faas-cli up -f example-site.yml
+```
+
+## Create a DNS A record for your sub-domain
+I'm using my personal domain `matiaspan.dev` that I got at [namecheap.com](https://namecheap.com). I can create sub-domains for each function. If you are familiar with DNS, then create an A record for the IP address of the LoadBalancer created for the ingress controller(i.e nginx):
+```sh
+kubectl get svc -n default
+NAME                                         TYPE           CLUSTER-IP       EXTERNAL-IP      PORT(S)                      AGE
+nginxingress-nginx-ingress-controller        LoadBalancer   10.245.116.147   159.89.221.240   80:31113/TCP,443:30596/TCP   2d21h
+```
+
+You can see that my external IP address is: `159.89.221.240`.  
+If you are getting started with DNS management I recommend [DigitalOcean](https://digitalocean.com), it's free and has a great developer experience. Here is how I use the CLI to create the record for `example.matiaspan.dev`:
```

Comment:
I learnt that when writing blog posts with example domain names, you should always use the example.com domain, it's specifically set up for this purpose.

Let's change: `example.matiaspan.dev` to `example.com` or `my-site.example.com`

Reply:
At first I used `example.com`, but then I noticed that in the nginx blog post you were using real domains(i.e `my-homepage.myfaas.club`) so I decided to go with the real domain I used

Reply:
So I would say - use either `my-site.example.com` or `my-site.matiaspan.dev`

Reply:
No problem